tools/toolchain/clang_lto_pgo.sh

```diff
@@ -0,0 +1,143 @@
+    exit 0
+elif [ "${CLANG_BUILD}" = "BUILD" ]; then
+    echo "build optimized clang"
+elif [ "${CLANG_BUILD}" = "INSTALL" ]; then
+    echo "build and install optimized clang"
+elif [ "${CLANG_BUILD}" = "INSTALL_PREBUILT" ]; then
+    echo "install prebuilt optimized clang"
+else
+    echo "Not sure what to do with ${CLANG_BUILD}"
+    exit 1
+fi
+
+if [ ! -d "/optimized_clang" ]; then
+    echo "/optimized_clang not mounted"
+    exit 1
+fi
+DIR="/optimized_clang"
+
+STAGE0_BIN=$DIR/stage-0-${CLANG_ARCH}/build/bin
+STAGE1_BIN=$DIR/stage-1-${CLANG_ARCH}/build/bin
+
+# Which LLVM release to build in order to compile Scylla
+LLVM_SCYLLA_TAG=17.0.6
+
+# Which LLVM release to use to build clang (stage 0).
+LLVM_CLANG_TAG=17.0.6
+
+if [ "${CLANG_BUILD}" = "INSTALL" -o "${CLANG_BUILD}" = "BUILD" ]; then
+    cd "${DIR}"
+    # Clone, patch and bootstrap the newest Clang.
+    rm -rf stage-0-${CLANG_ARCH}
+    git clone https://github.com/llvm/llvm-project --branch llvmorg-${LLVM_CLANG_TAG} --depth=1 stage-0-${CLANG_ARCH}
+    cd stage-0-${CLANG_ARCH}
+    USE_CURRENT_COMPILER=(-DCMAKE_C_COMPILER="/usr/bin/clang" -DCMAKE_CXX_COMPILER="/usr/bin/clang++" -DLLVM_USE_LINKER="/usr/bin/ld")
+    COMMON_OPTS=(-DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD=all -DLLVM_TARGET_ARCH=${LLVM_TARGET_ARCH} -G Ninja -DLLVM_INCLUDE_BENCHMARKS=OFF -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF -DLLVM_ENABLE_BINDINGS=OFF)
+    SCYLLA_OPTS=(--date-stamp $(date "+%Y%m%d") --debuginfo 1 --tests-debuginfo 1 --c-compiler="$STAGE1_BIN/clang" --compiler="$STAGE1_BIN/clang++")
+
+    echo "stage-0: build the bootstrapped compiler"
+
+    cmake -B build -S llvm  "${USE_CURRENT_COMPILER[@]}" "${COMMON_OPTS[@]}" "${LLVM_TARGET}" -DLLVM_ENABLE_PROJECTS="clang;lld;bolt" -DLLVM_ENABLE_RUNTIMES="compiler-rt" -DCOMPILER_RT_BUILD_SANITIZERS=OFF -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON
+    ninja -C build
+
+    USE_NEW_COMPILER=(-DCMAKE_C_COMPILER="$STAGE0_BIN/clang" -DCMAKE_CXX_COMPILER="$STAGE0_BIN/clang++" -DLLVM_USE_LINKER="$STAGE0_BIN/ld.lld")
+    NEW_COMMON_OPTS=(-DLLVM_ENABLE_PROJECTS="clang;lld" -DLLVM_ENABLE_RUNTIMES="compiler-rt" -DLLVM_ENABLE_LTO=Thin -DCLANG_DEFAULT_PIE_ON_LINUX=OFF -DLLVM_BUILD_TOOLS=OFF -DCMAKE_C_FLAGS="-Xclang -mllvm -Xclang -vp-counters-per-site=16" -DCMAKE_CXX_FLAGS="-Xclang -mllvm -Xclang -vp-counters-per-site=16")
+
+    echo "stage-1: Build a PGO-optimized compiler using the boostrapped compiler"
+
+    # Build a PGO-optimized compiler using the boostrapped compiler.
+    rm -rf ../stage-1-${CLANG_ARCH}
+    git fetch --depth=1 origin tag llvmorg-${LLVM_SCYLLA_TAG}
+    git worktree add ../stage-1-${CLANG_ARCH} llvmorg-${LLVM_SCYLLA_TAG}
+    cd ../stage-1-${CLANG_ARCH}
+    cmake -B build -S llvm "${USE_NEW_COMPILER[@]}" "${COMMON_OPTS[@]}" "${NEW_COMMON_OPTS[@]}" ${LLVM_TARGET} -DLLVM_BUILD_INSTRUMENTED=IR
+    ninja -C build
+
+    echo "cmake stage-1 output:"
+
+    cd ../
+    # Download the scylla codebase for training.
```

Comment:
where's the download?

Reply:
No need, use this tree.

Reply:
It's bring in current source tree. I need to change this comment.